src/Jab.FunctionalTests.Common/Mocks/ServiceImplementationWithNullable.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+
+namespace JabTests;
+
+#nullable enable
+
+internal class ServiceImplementationWithNullable : IService
+{
+    public IService1 Parameter1 { get; }
+    public IService2? Parameter2 { get; }
+    public IEnumerable<IService3>? Parameter3 { get; }
+
+    public ServiceImplementationWithNullable(
+        IService1 parameter1,
+        IService2? parameter2,
```

Comment:
Should we be making the decision based on nullable annotation? Feels not explicit enough. 

What about still requiring the default value for nullable parameters? And maybe emitting a warning for nullable required parameters?

Reply:
I agree and making some changes. The problem is that the types `LoggerFactory` and `ConsoleLoggerProvider` uses nullable parameters for optional and will raise warnings.

Reply:
Changed to raise a warning